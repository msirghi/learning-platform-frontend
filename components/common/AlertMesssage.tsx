import React from 'react';
import Alert from '@material-ui/lab/Alert';
import { AlertType } from '../../common/enums';

/**
 * AlertMessage component props.
 */
type Props = {
  /**
   * Message that will be displayed
   */
  message: string;
  /**
   * Alert type. Depending on it component will have special colors.
   */
  type: AlertType;
};

/**
 * Component for displaying the alert messages. Uses Alert component from '@material-ui/lab' library.
 *
 * @version 0.1
 * @author [Sirghi Mihail](https://github.com/msirghi)
 */
export const AlertMessage: React.FC<Props> = ({ message, type }) => {
  return <Alert severity={type}>{message}</Alert>;
};
