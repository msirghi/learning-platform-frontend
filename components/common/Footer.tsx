import React from 'react';
import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import Container from '@material-ui/core/Container';
import Link from '@material-ui/core/Link';

function Copyright() {
  return (
    <Typography variant='body2'>
      {'Copyright Â© '}
      <Link color='inherit' href='https://material-ui.com/'>
        Your Website
      </Link>{' '}
      {new Date().getFullYear()}
      {'.'}
    </Typography>
  );
}

const useStyles = makeStyles((theme) => ({
  footer: {
    padding: theme.spacing(3, 2),
    marginTop: 'auto',
    backgroundColor: '#202020',
    color: '#fff'
  }
}));

export const SiteFooter = () => {
  const classes = useStyles();
  return (
    <footer className={classes.footer}>
      <Container maxWidth='sm'>
        <Typography variant='body1'>My sticky footer can be found here.</Typography>
        <Copyright />
      </Container>
    </footer>
  );
};
