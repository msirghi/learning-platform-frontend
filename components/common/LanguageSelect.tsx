import React, { useState } from 'react';
import LanguageIcon from '@material-ui/icons/Language';
import IconButton from '@material-ui/core/IconButton';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Menu from '@material-ui/core/Menu';
import { useTranslation } from '../../i18n';
import { useDispatch } from 'react-redux';
import { setInterfaceLocale } from '../../redux/actions/preference/preferenceAction';

const languageChoice = [
  {
    key: 'en',
    value: 'English'
  },
  {
    key: 'ru',
    value: 'Русский'
  }
];

export const LanguageSelect: React.FC = () => {
  const [notificationMenuOpen, setNotificationMenuOpen] = useState<null | HTMLElement>(null);
  const { i18n } = useTranslation();
  const dispatch = useDispatch();

  const handleChange = (val: string) => {
    dispatch(setInterfaceLocale(val));
    i18n.changeLanguage(val);
    handleClose();
  };

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setNotificationMenuOpen(event.currentTarget);
  };

  const handleClose = () => {
    setNotificationMenuOpen(null);
  };

  return (
    <>
      <IconButton onClick={handleClick}>
        <LanguageIcon color='primary' />
      </IconButton>
      <Menu
        anchorEl={notificationMenuOpen}
        keepMounted
        open={Boolean(notificationMenuOpen)}
        onClose={handleClose}
      >
        <List>
          {languageChoice.map((item) => {
            return (
              <ListItem key={item.key} button onClick={() => handleChange(item.key)}>
                {item.value}
              </ListItem>
            );
          })}
        </List>
      </Menu>
    </>
  );
};
