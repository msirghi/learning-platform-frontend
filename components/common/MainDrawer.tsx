import React, { useEffect, useState } from "react";
import Drawer from "@material-ui/core/Drawer";
import IconButton from "@material-ui/core/IconButton";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import { makeStyles } from "@material-ui/core/styles";
import clsx from "clsx";
import { menuItems } from "./utils/menuItems";
import styles from "../../styles/modules/Shared.module.scss";
import { Hidden } from "@material-ui/core";

const drawerWidth = 240;

const useStyles = makeStyles((theme) => ({
  drawer: {
    width: drawerWidth,
  },
  drawerOpen: {
    background: theme.palette.primary.main,
    width: drawerWidth,
    transition: theme.transitions.create("width", {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.enteringScreen,
    }),
  },
  drawerClose: {
    background: theme.palette.primary.main,
    transition: theme.transitions.create("width", {
      easing: theme.transitions.easing.sharp,
      duration: theme.transitions.duration.leavingScreen,
    }),
    overflowX: "hidden",
    width: theme.spacing(8) + 1,
    [theme.breakpoints.up("sm")]: {
      width: 110,
    },
  },
  toolbar: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    padding: theme.spacing(0, 1),
    ...theme.mixins.toolbar,
  },
  drawerPaper: {
    width: drawerWidth,
    background: theme.palette.primary.main,
  },
}));

type Props = {
  handleDrawerClose: () => void;
  open: boolean;
  mobileOpen: boolean;
};

export const MainDrawer: React.FC<Props> = ({ handleDrawerClose, open, mobileOpen }) => {
  const classes = useStyles();
  const [activeTab, setActiveTab] = useState<string | null>(null);

  useEffect(() => {
    /* istanbul ignore next */
    if (window) {
      setActiveTab(window.location.href);
    }
  }, []);

  if (!activeTab) {
    return <div />;
  }

  const drawerContent = () => {
    return (
      <>
        <div className={classes.toolbar}>
          <IconButton onClick={handleDrawerClose}>
            <img src='/images/register-logo-148.png' height={72} width={72} />
          </IconButton>
        </div>
        <List className={styles.mainMenuListContainer}>
          {menuItems.map((item) => {
            const isActiveTab = activeTab.includes(item.link);
            return (
              <div
                className={`${styles.mainMenuColor} ${isActiveTab && styles.mainMenuActive} ${
                  item.className
                }`}
                key={item.label}
              >
                <ListItem button>
                  <ListItemIcon className={styles.mainMenuIconContainer}>
                    <item.icon
                      className={`${isActiveTab ? styles.mainMenuActiveIcon : styles.mainMenuIcon}`}
                    />
                  </ListItemIcon>
                  {(open || mobileOpen) && (
                    <ListItemText
                      className={isActiveTab ? styles.mainMenuActiveText : ""}
                      primary={item.label}
                    />
                  )}
                </ListItem>
              </div>
            );
          })}
        </List>
      </>
    );
  };

  return (
    <>
      <Drawer
        variant='temporary'
        open={mobileOpen}
        onClose={handleDrawerClose}
        classes={{
          paper: classes.drawerPaper,
        }}
        ModalProps={{
          keepMounted: true,
        }}
      >
        {drawerContent()}
      </Drawer>
      <Hidden xsDown implementation='css'>
        <Drawer
          classes={{
            paper: clsx({
              [classes.drawerOpen]: open,
              [classes.drawerClose]: !open,
            }),
          }}
          className={clsx(classes.drawer, {
            [classes.drawerOpen]: open,
            [classes.drawerClose]: !open,
          })}
          variant='permanent'
          open={open}
        >
          {drawerContent()}
        </Drawer>
      </Hidden>
    </>
  );
};
