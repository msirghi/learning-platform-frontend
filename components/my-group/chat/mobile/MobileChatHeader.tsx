import React from 'react';
import styles from '../../../../styles/modules/MyGroup.module.scss';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import { IconButton } from '@material-ui/core';
import SettingsIcon from '@material-ui/icons/Settings';

type Props = {
  onBackClick: () => void;
  toggleSettings: () => void;
  online: number;
  total: number;
};

export const MobileChatHeader: React.FC<Props> = ({
  onBackClick,
  toggleSettings,
  online,
  total
}) => {
  return (
    <div className={styles.mobileHeader}>
      <div>
        <IconButton onClick={() => onBackClick()}>
          <ArrowBackIcon className={styles.backIcon} />
        </IconButton>
      </div>
      <div>
        <div className={styles.groupName}>Group name</div>
        <div className={styles.stats}>
          <span>{total} members</span>
          {online > 0 && <span> / {online} online</span>}
        </div>
      </div>
      <div className={styles.settingsIcon}>
        <IconButton onClick={() => toggleSettings()}>
          <SettingsIcon />
        </IconButton>
      </div>
    </div>
  );
};
