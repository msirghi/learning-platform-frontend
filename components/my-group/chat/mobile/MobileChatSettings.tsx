import React from 'react';
import { ColorPaletteTemplate } from '../settings/ColorPaletteTemplate';
import styles from '../../../../styles/modules/MyGroup.module.scss';
import { chatThemes } from '../../../common/utils/chatThemes';
import { FontSizeSlider } from '../settings/FontSizeSlider';
import { Button } from '@material-ui/core';
import { ChatThemes } from '../../../../common/enums';

type Props = {
  currentColor: string;
  chatFontSize: number;
  handleThemeChange: (pallete: ChatThemes) => void;
  handleFontChange: (val: number) => void;
  toggleSettings: () => void;
};

export const MobileChatSettings: React.FC<Props> = ({
  currentColor,
  chatFontSize,
  handleThemeChange,
  handleFontChange,
  toggleSettings
}) => {
  return (
    <div className={styles.mobileChatSettingsContainer}>
      <div className={styles.mainTitle}>Chat settings</div>
      <div className={styles.themeTitle}>Theme</div>
      <div className={styles.themeSubtitle}>Select a theme for your chat</div>
      <div className={styles.mobileChatPalettes}>
        {chatThemes.map(({ color }) => {
          return (
            <div className={styles.theme} key={color}>
              <ColorPaletteTemplate
                onChange={() => handleThemeChange(color as ChatThemes)}
                color={color}
                checked={color === currentColor}
              />
            </div>
          );
        })}
      </div>

      <div className={styles.fontSizeTitle}>Font size</div>
      <div className={styles.fontSizeSubtitle}>Adjust the chat font size</div>
      <FontSizeSlider value={chatFontSize} handleFontChange={(val) => handleFontChange(val)} />

      <div className={styles.applyButtonContainer}>
        <Button color='primary' variant='contained' fullWidth onClick={() => toggleSettings()}>
          Apply
        </Button>
      </div>
    </div>
  );
};
