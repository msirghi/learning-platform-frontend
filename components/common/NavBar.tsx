import React, { useEffect, useState } from 'react';
import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Toolbar from '@material-ui/core/Toolbar';
import clsx from 'clsx';
import Typography from '@material-ui/core/Typography';
import NotificationsIcon from '@material-ui/icons/Notifications';
import styles from '../../styles/modules/Shared.module.scss';
import Divider from '@material-ui/core/Divider';
import { UserAvatar } from './UserAvatar';
import Badge from '@material-ui/core/Badge';
import ChevronLeftIcon from '@material-ui/icons/ChevronLeft';
import { LanguageSelect } from './LanguageSelect';
import { useTranslation } from '../../i18n';

type Props = {
  classes: { [key: string]: string };
  handleDrawerOpen: () => void;
  handleDrawerClose: () => void;
  open: boolean;
};

export const NavBar: React.FC<Props> = ({ classes, handleDrawerOpen, open, handleDrawerClose }) => {
  const [activeTab, setActiveTab] = useState<string | undefined>();
  const { t, i18n } = useTranslation();

  useEffect(() => {
    if (global.window) {
      console.log('global.window.location :>> ', global.window.location.pathname);
      const href = global.window.location.pathname.replaceAll('/', '');
      setActiveTab(t(`common:headerTabs.${href}`));
    }
  }, [global.window, i18n]);

  return (
    <AppBar
      position='fixed'
      className={clsx(classes.appBar, {
        [classes.appBarShift]: open
      })}
    >
      <Toolbar>
        <IconButton
          color='inherit'
          aria-label='open drawer'
          onClick={handleDrawerOpen}
          edge='start'
          className={clsx(classes.menuButton, {
            [classes.hide]: open
          })}
        >
          <MenuIcon />
        </IconButton>

        <IconButton
          color='inherit'
          aria-label='close drawer'
          onClick={handleDrawerClose}
          edge='start'
          className={clsx(classes.menuButton, {
            [classes.hide]: !open
          })}
        >
          <ChevronLeftIcon />
        </IconButton>
        <Typography variant='h6' noWrap>
          {activeTab}
        </Typography>
        <div className={styles.navBarRightIconsContainer}>
          <LanguageSelect />
          <IconButton>
            <Badge badgeContent={4} color='primary'>
              <NotificationsIcon color={'primary'} />
            </Badge>
          </IconButton>
          <IconButton>
            <UserAvatar />
          </IconButton>
        </div>
      </Toolbar>
      <div className={styles.navBarDividerContainer}>
        <Divider light />
      </div>
    </AppBar>
  );
};
