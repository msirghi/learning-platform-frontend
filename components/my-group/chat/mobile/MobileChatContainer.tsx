import React, { useEffect, useRef, useState } from 'react';
import { messagesMock } from '../../../../mocks/messagesMocks';
import { Message } from '../Message';
import styles from '../../../../styles/modules/MyGroup.module.scss';
import { ChatPrimaryTextField } from '../ChatPrimaryTextField';
import { MobileChatHeader } from './MobileChatHeader';
import { Fade } from '@material-ui/core';
import { MobileChatSettings } from './MobileChatSettings';
import { ChatThemes } from '../../../../common/enums';

type Props = {
  onBackClick: () => void;
  chatOpened: boolean;
  chatTheme: ChatThemes;
  handleThemeChange: (val: ChatThemes) => void;
  chatFontSize: number;
  handleFontChange: (val: number) => void;
};

export const MobileChatContainer: React.FC<Props> = ({
  onBackClick,
  chatOpened,
  handleThemeChange,
  chatTheme,
  chatFontSize,
  handleFontChange
}) => {
  const [messageValue, setMessageValue] = useState('');
  const [messages, setMessages] = useState(messagesMock);
  const [chatSettingsOpened, setChatSettingsOpened] = useState(false);
  const messagesEnd = useRef(null);

  const onInputChange = (val: string) => {
    setMessageValue(val);
  };

  const onMessageSend = () => {
    if (messageValue) {
      setMessages((m) => [
        ...m,
        { id: Math.random(), author: 'Me', date: new Date(), message: messageValue }
      ]);
      setMessageValue('');
    }
  };

  const onKeyDown = (e: React.KeyboardEvent<HTMLDivElement>) => {
    if (e.keyCode == 13) {
      onMessageSend();
    }
  };

  const scrollToLastMessage = () => {
    messagesEnd.current.scrollTop = messagesEnd.current.scrollHeight;
  };

  const onFocus = () => {
    scrollToLastMessage();
  };

  useEffect(() => scrollToLastMessage(), []);

  useEffect(() => {
    scrollToLastMessage();
  }, [messages]);

  return (
    <div className={styles.mobileChatContent}>
      <MobileChatHeader
        online={8}
        total={1}
        onBackClick={onBackClick}
        toggleSettings={() => setChatSettingsOpened(!chatSettingsOpened)}
      />
      {chatSettingsOpened ? (
        <MobileChatSettings
          chatFontSize={chatFontSize}
          handleFontChange={handleFontChange}
          currentColor={chatTheme}
          handleThemeChange={(val) => handleThemeChange(val)}
          toggleSettings={() => setChatSettingsOpened(!chatSettingsOpened)}
        />
      ) : (
        <Fade in={chatOpened}>
          <div className={styles.fixedContainer} style={{ backgroundColor: chatTheme }}>
            <div
              className={styles.mobileChatContainer}
              style={{ backgroundColor: chatTheme }}
              ref={messagesEnd}
            >
              {messages.map((message, idx) => {
                return (
                  <>
                    <Message
                      fontSize={chatFontSize}
                      hideAvatar
                      fromMe={idx % 2 === 0}
                      chatMessage={message}
                      key={message.id}
                    />
                    <div
                      className={`${
                        idx % 2 === 0 ? styles.myMessageMobile : styles.notMyMessageMobile
                      }`}
                    >
                      {message.author}
                    </div>
                  </>
                );
              })}
            </div>
            <div className={styles.mobileTextFieldContainer}>
              <ChatPrimaryTextField
                onFocus={onFocus}
                onKeyDown={onKeyDown}
                onChange={onInputChange}
                onMessageSend={onMessageSend}
                value={messageValue}
              />
            </div>
          </div>
        </Fade>
      )}
    </div>
  );
};
