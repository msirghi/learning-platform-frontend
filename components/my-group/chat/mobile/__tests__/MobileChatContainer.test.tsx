import { fireEvent, render } from '@testing-library/react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import { ChatThemes } from '../../../../../common/enums';
import { MobileChatContainer } from '../MobileChatContainer';
import { MobileChatHeader } from '../MobileChatHeader';
import { MobileChatSettings } from '../MobileChatSettings';
import userEvent from '@testing-library/user-event';

describe('MobileChatContainer component', () => {
  const props = {
    onBackClick: jest.fn(),
    chatOpened: false,
    chatTheme: ChatThemes.WHITE,
    handleThemeChange: jest.fn(),
    chatFontSize: 14,
    handleFontChange: jest.fn()
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot on valid props', () => {
    const wrapper = shallow(<MobileChatContainer {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should toggle mobile chat settings on header click', () => {
    const wrapper = shallow(<MobileChatContainer {...props} />);
    const header = wrapper.find(MobileChatHeader);

    expect(header).toHaveLength(1);
    header.props().toggleSettings();
    wrapper.update();
    const settings = wrapper.find(MobileChatSettings);
    expect(settings).toHaveLength(1);
  });

  it('should handle theme change on for mobile devices', () => {
    const spy = jest.fn((data) => data);
    const wrapper = shallow(<MobileChatContainer {...props} handleThemeChange={spy} />);
    const header = wrapper.find(MobileChatHeader);
    header.props().toggleSettings();
    wrapper.update();

    const settings = wrapper.find(MobileChatSettings);
    expect(settings).toHaveLength(1);
    settings.props().handleThemeChange(ChatThemes.WHEAT);
    expect(spy).toBeCalledTimes(1);
    expect(spy).toBeCalledWith(ChatThemes.WHEAT);
  });

  it('should handle settings toggle on for mobile devices', () => {
    const wrapper = shallow(<MobileChatContainer {...props} />);
    const header = wrapper.find(MobileChatHeader);
    header.props().toggleSettings();
    wrapper.update();

    const settings = wrapper.find(MobileChatSettings);
    expect(settings).toHaveLength(1);
    settings.props().toggleSettings();
    wrapper.update();
    expect(wrapper.find('.fixedContainer')).toHaveLength(1);
  });

  it('should handle text field change', () => {
    const { queryByTestId } = render(<MobileChatContainer {...props} />);
    const input = queryByTestId('message-field');
    expect(input).toBeInTheDocument();

    fireEvent.change(input, { target: { value: 'new value' } });
    expect(input).toHaveValue('new value');
    fireEvent.keyDown(input, { keyCode: 13 });
    expect(input).toHaveValue('');
  });

  it('should handle text field focus', () => {
    const { queryByTestId } = render(<MobileChatContainer {...props} />);
    const input = queryByTestId('message-field');
    expect(input).toBeInTheDocument();

    userEvent.click(input);
    expect(input).toHaveFocus();
  });

  it('should handle text field keydown', () => {
    const { queryByTestId } = render(<MobileChatContainer {...props} />);
    const input = queryByTestId('message-field');
    expect(input).toBeInTheDocument();

    userEvent.type(input, 'new value');
    expect(input).toHaveValue('new value');
    fireEvent.keyDown(input, { keyCode: 13 });

    const sendButton = queryByTestId('send-button');
    expect(sendButton).toBeInTheDocument();
    userEvent.click(sendButton);
    expect(input).toHaveValue('');
  });

  it('should not clear the field if enter was not pressed', () => {
    const { queryByTestId } = render(<MobileChatContainer {...props} />);
    const input = queryByTestId('message-field');
    expect(input).toBeInTheDocument();

    userEvent.type(input, 'new value');
    expect(input).toHaveValue('new value');
    fireEvent.keyDown(input, { keyCode: 14 });
    expect(input).toHaveValue('new value');
  });

  it('should render the field with the same value on trying to clear the empty value', () => {
    const { queryByTestId } = render(<MobileChatContainer {...props} />);
    const input = queryByTestId('message-field');
    expect(input).toBeInTheDocument();

    userEvent.type(input, '');
    expect(input).toHaveValue('');

    const sendButton = queryByTestId('send-button');
    expect(sendButton).toBeInTheDocument();
    userEvent.click(sendButton);
    expect(input).toHaveValue('');
  });
});
