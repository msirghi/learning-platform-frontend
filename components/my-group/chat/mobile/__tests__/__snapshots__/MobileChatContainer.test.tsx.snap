// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileChatContainer component should match the snapshot on valid props 1`] = `
<div
  className="mobileChatContent"
>
  <MobileChatHeader
    onBackClick={[MockFunction]}
    online={8}
    toggleSettings={[Function]}
    total={1}
  />
  <ForwardRef(Fade)
    in={false}
  >
    <div
      className="fixedContainer"
      style={
        Object {
          "backgroundColor": "#fafafa",
        }
      }
    >
      <div
        className="mobileChatContainer"
        style={
          Object {
            "backgroundColor": "#fafafa",
          }
        }
      >
        <Message
          chatMessage={
            Object {
              "author": "Mihail",
              "date": 2000-11-22T00:00:00.000Z,
              "id": 1,
              "message": "Lorem ipsum dolor sit amet.",
            }
          }
          fontSize={14}
          fromMe={true}
          hideAvatar={true}
          key="1"
        />
        <div
          className="myMessageMobile"
        >
          Mihail
        </div>
        <Message
          chatMessage={
            Object {
              "author": "Nicolae",
              "date": 2000-11-22T00:00:00.000Z,
              "id": 2,
              "message": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Praesentium, sapiente!",
            }
          }
          fontSize={14}
          fromMe={false}
          hideAvatar={true}
          key="2"
        />
        <div
          className="notMyMessageMobile"
        >
          Nicolae
        </div>
        <Message
          chatMessage={
            Object {
              "author": "Mihail 2",
              "date": 2000-11-22T00:00:00.000Z,
              "id": 3,
              "message": "Lorem ipsum, dolor sit amet consectetur adipisicing.",
            }
          }
          fontSize={14}
          fromMe={true}
          hideAvatar={true}
          key="3"
        />
        <div
          className="myMessageMobile"
        >
          Mihail 2
        </div>
        <Message
          chatMessage={
            Object {
              "author": "Mihail 4",
              "date": 2000-11-22T00:00:00.000Z,
              "id": 4,
              "message": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!",
            }
          }
          fontSize={14}
          fromMe={false}
          hideAvatar={true}
          key="4"
        />
        <div
          className="notMyMessageMobile"
        >
          Mihail 4
        </div>
        <Message
          chatMessage={
            Object {
              "author": "Mihail 4",
              "date": 2000-11-22T00:00:00.000Z,
              "id": 5,
              "message": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!",
            }
          }
          fontSize={14}
          fromMe={true}
          hideAvatar={true}
          key="5"
        />
        <div
          className="myMessageMobile"
        >
          Mihail 4
        </div>
        <Message
          chatMessage={
            Object {
              "author": "Mihail 4",
              "date": 2000-11-22T00:00:00.000Z,
              "id": 6,
              "message": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Accusamus, repudiandae? Iusto labore dolores obcaecati maxime. Quaerat praesentium doloribus expedita voluptatum!",
            }
          }
          fontSize={14}
          fromMe={false}
          hideAvatar={true}
          key="6"
        />
        <div
          className="notMyMessageMobile"
        >
          Mihail 4
        </div>
      </div>
      <div
        className="mobileTextFieldContainer"
      >
        <ChatPrimaryTextField
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMessageSend={[Function]}
          value=""
        />
      </div>
    </div>
  </ForwardRef(Fade)>
</div>
`;
